backend/entityservice/tasks/comparing.py

```diff
@@ -148,10 +148,11 @@ def compute_filter_similarity(chunk_info_dp1, chunk_info_dp2, project_id, run_id
     span.log_kv({'event': 'chunks are fetched and deserialized'})
     log.debug("Calculating filter similarity")
     span.log_kv({'size1': chunk_dp1_size, 'size2': chunk_dp2_size})
-    chunk_results = anonlink.entitymatch.calculate_filter_similarity(chunk_dp1, chunk_dp2,
-                                                                     threshold=threshold,
-                                                                     k=min(chunk_dp1_size, chunk_dp2_size),
-                                                                     use_python=False)
+    chunk_results = anonlink.candidate_generation.find_candidate_pairs(
+        (chunk_dp1, chunk_dp2),
+        anonlink.similarities.dice_coefficient_accelerated,
+        threshold,
+        k=min(chunk_dp1_size, chunk_dp2_size))
```

Comment:
Recalling our conversation last week about `k` restricting the number of similarity scores to the size of the smaller party which is a problem when outputting similarity scores directly. 

Do you see any issue with changing this to `max`? Probably makes more sense to do in a subsequent issue/PR but just wanted to flag it.

Reply:
The only possible issue would be performance if the two datasets are of very different size. It would change the time complexity from _O_(min(_n_, _m_)²) to _O_(_nm_).

Reply:
Also, I’ve historically been in favour of just getting rid of this _k_ parameter from our code altogether. It’s meant to prevent the similarity computation from taking up too much memory (when there’s too many pairs above the threshold) but we don’t actually use it anywhere. Further, as it’s currently implemented, it’s not symmetric, so it causes you to get different results if you swap the order of the dataset. Also, it was a serious pain to implement maintaining only _k_ closest matches for a record when the similarity computation is distributed (nb: this is implemented in anonlink; still needs to be called from ES).